src/pybind/mgr/cephadm/migrations.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from typing import List, TYPE_CHECKING, Iterator
+
+from ceph.deployment.service_spec import PlacementSpec, ServiceSpec, HostPlacementSpec
+from cephadm.inventory import Inventory, SpecStore, HostCache
+from cephadm.schedule import HostAssignment
+
+from orchestrator import DaemonDescription, OrchestratorError
+
+if TYPE_CHECKING:
+    from .module import CephadmOrchestrator
+
+LAST_MIGRATION = 1
+
+logger = logging.getLogger(__name__)
+
+
+
+class Migrations:
+    def __init__(self, mgr: "CephadmOrchestrator"):
+        self.mgr = mgr
+
+        self.initialize()
+
+    def initialize(self):
+        if self.mgr.migration_current is None:
+            self.set(0)
+        self.migrate()
```

Comment:
start the migration during both `__init__()` and `serve()` ?

Reply:
just settting `migration_current = 1` is often the only thing that needs to be done. That's why I also do that in __init__.

Reply:
added explanation